src/nvim/ops.c

```diff
@@ -815,9 +815,17 @@ yankreg_T *get_yank_register(int regname, int mode)
 {
   yankreg_T *reg;
 
-  if (mode == YREG_PASTE && get_clipboard(regname, &reg, false)) {
+  if ((mode == YREG_PASTE || mode == YREG_PUT)
+      && get_clipboard(regname, &reg, false)) {
     // reg is set to clipboard contents.
     return reg;
+  } else if (mode == YREG_PUT && (regname == '*' || regname == '+')) {
+    // in case clipboard not available and we aren't actually pasting,
+    // return an empty register
+    reg = xmalloc(sizeof(yankreg_T));
+    reg->y_array = NULL;
+    reg->y_size = 0;
```

Comment:
Hmm, the register returned by this function should not be allocated, so it has to be a pointer to a static variable instead.
```suggestion
    static yankreg_T empty_reg = { .y_array = NULL };
    reg = &empty_reg;
```